### DISCUSSION TITLE
server_onupdate ignored when generating migration script

### DISCUSSION THREAD

**Discussion by TheCaffinatedDeveloper** [sqlalchemy/alembic#1055] [Usage Questions] [answered]
**Describe the bug**

Alembic is ignoring server_onupdate column when generating migration file.
```
modified_on = Column(DateTime, nullable=False, server_default=func.now(), server_onupdate=func.current_timestamp())
```
The generated migration will be
```
sa.Column('modified_on', sa.DateTime(), server_default=sa.text('now()'), nullable=False)
```

**Expected behavior**

Expected
```
sa.Column('modified_on', sa.DateTime(), server_default=sa.text('now()'), server_onupdate=sa.text('now()'), nullable=False)
```

**To Reproduce**
Please try to provide a [Minimal, Complete, and Verifiable](http://stackoverflow.com/help/mcve) example, with the migration script and/or the SQLAlchemy tables or models involved.
See also [Reporting Bugs](https://www.sqlalchemy.org/participate.html#bugs) on the website.

No errors generated and producing simply requires generating a migration file and looking at the `modified_on` column.

**Versions.**
 - OS: MacOS 12.1
 - Python: 3.8.9
 - Alembic: 1.7.6
 - SQLAlchemy: 1.4.31
 - Database: Postgres 12
 - DBAPI: N/A

**Additional context**

**Have a nice day!**

**Comment by zzzeek** [comment, marked as answer]
server_onupdate is a Python behavioral element only and has no effect on DDL, which is the purpose of alembic.  how is this affecting your use case?

**Comment by TheCaffinatedDeveloper** [comment]
Thanks @zzzeek , this was a confusing point since this behaves different than server_default. I reread the documentation and it does say that it does not have an effect on DDL. Out of curiosity, is it possible to create the necessary trigger function to add this functionality?

**Comment by zzzeek** [comment]
triggers are very specific to each database vendor, they are written in a style that is more imperative rather than declarative (meaning, you really have to use the trigger syntax directly to be effective) and they tend to introduce lots of complications to things (changing the behaviors of INSERT/UPDATE/DELETE in many idiosyncratic ways), so SQLAlchemy/ Alembic steers completely clear of having any direct interaction with trigger syntax.   

however good news is you're on PostgreSQL and there's a great project [alembic-utils](https://olirice.github.io/alembic_utils/) where someone was nice enough to get into all of that, so perhaps this project can provide some help on adding triggers.
